pilot/pkg/xds/ads.go

```diff
@@ -394,6 +394,14 @@ func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.Disc
 		log.Debugf("ADS:%s: INIT/RECONNECT %s %s %s", stype, con.conID, request.VersionInfo, request.ResponseNonce)
 		con.proxy.Lock()
 		con.proxy.WatchedResources[request.TypeUrl] = &model.WatchedResource{TypeUrl: request.TypeUrl, ResourceNames: request.ResourceNames}
+		// Due to https://github.com/envoyproxy/envoy/issues/13009, we must always send an EDS response when CDS is pushed.
+		// However, during reconnect scenarios, Envoy may fail to send EDS requests that we can reasonably identify as non-ACKs.
```

Comment:
I understand when reconnect, the connection  between envoy->xds proxy  will be closed too.  Then envoy will start the xds requests as initially. How could this happen?